src/functions.jl

```diff
@@ -43,77 +43,140 @@ struct VectorOfVariables <: AbstractVectorFunction
 end
 
 """
-    ScalarAffineFunction{T}(variables, coefficients, constant)
+    struct ScalarAffineTerm{T}
+        coefficient::T
+        variable_index::VariableIndex
+    end
+
+Represents ``c x_i`` where ``c`` is `coefficient` and ``x_i`` is the variable
+identified by `variable_index`.
+"""
+struct ScalarAffineTerm{T}
+    coefficient::T
+    variable_index::VariableIndex
+end
+
+"""
+    ScalarAffineFunction{T}(terms, constant)
 
 The scalar-valued affine function ``a^T x + b``, where:
-* ``a`` is a sparse vector specified in tuple form by `variables::Vector{VariableIndex}` and `coefficients::Vector{T}`
+* ``a`` is a sparse vector specified by a list of
+  [`ScalarAffineTerm`](@ref ScalarAffineTerm) structs.
```

Comment:
Isn't
```julia
[`ScalarAffineTerm`](@ref)
```
sufficient ?

Reply:
It's a bit awkward to say
```
[`ScalarAffineTerm`](@ref)s
```
(with the s).

Reply:
I realize that this rule isn't applied consistently though.

Reply:
There is currently no "s", is there ?

Reply:
"a list of ScalarAffineTerm" is not really grammatical. It should be "a list of ScalarAffineTerms" or "a list of ScalarAffineTerm structs".

Reply:
Oops, my bad for misunderstanding. Yes, I'll change this.